src/Writers/LaTeXWriter.jl

```diff
@@ -13,12 +13,415 @@ import ...Documenter:
+    println(io, "\n\\hypertarget{", id, "}{}\n")
+    latex(io, anchor.object, page, doc)
+end
+
+
+## Documentation Nodes.
+
+function latex(io::IO, node::Documents.DocsNodes, page, doc)
+    for node in node.nodes
+        latex(io, node, page, doc)
+    end
+end
+
+function latex(io::IO, node::Documents.DocsNode, page, doc)
+    local id = string(hash(string(node.anchor.id)))
+    # Docstring header based on the name of the binding and it's category.
+    println(io, "\\hypertarget{", id, "}{} ")
+    print(io, "\\hyperlink{", id, "}{\\texttt{", latexesc(string(node.object.binding)), "}} ")
+    println(io, " -- {", Utilities.doccat(node.object), ".}\n")
+    # # Body. May contain several concatenated docstrings.
+    println(io, "\\begin{adjustwidth}{2em}{0pt}")
+    latexdoc(io, node.docstr, page, doc)
+    println(io, "\n\\end{adjustwidth}")
+end
+
+function latexdoc(io::IO, md::Markdown.MD, page, doc)
+    if haskey(md.meta, :results)
+        # The `:results` field contains a vector of `Docs.DocStr` objects associated with
+        # each markdown object. The `DocStr` contains data such as file and line info that
+        # we need for generating correct scurce links.
+        for (markdown, result) in zip(md.content, md.meta[:results])
+            latex(io, Writers.MarkdownWriter.dropheaders(markdown), page, doc)
+            # When a source link is available then print the link.
+            Utilities.unwrap(Utilities.url(doc.internal.remote, doc.user.repo, result)) do url
+                link = "\\href{$url}{\\texttt{source}}"
+                println(io, "\n", link, "\n")
+            end
+        end
+    else
+        # Docstrings with no `:results` metadata won't contain source locations so we don't
+        # try to print them out. Just print the basic docstring.
+        render(io, mime, dropheaders(md), page, doc)
+    end
+end
+
+function latexdoc(io::IO, other, page, doc)
+    # TODO: properly support non-markdown docstrings at some point.
+    latex(io, other, page, doc)
+end
+
+
+## Index, Contents, and Eval Nodes.
+
+function latex(io::IO, index::Documents.IndexNode, page, doc)
+    println(io, "\\begin{itemize}")
+    for (object, _, page, mod, cat) in index.elements
+        local id = string(hash(string(Utilities.slugify(object))))
+        local text = latexesc(string(object.binding))
+        println(io, "\\item \\hyperlink{", id, "}{\\texttt{", text, "}}")
+    end
+    println(io, "\\end{itemize}\n")
+end
+
+function latex(io::IO, contents::Documents.ContentsNode, page, doc)
+    local depth = 1
+    local needs_end = false
+    println(io, "\\begin{itemize}")
+    for (count, path, anchor) in contents.elements
+        local header = anchor.object
+        local level = Utilities.header_level(header)
+        local id = string(hash(string(anchor.id, "-", anchor.nth)))
+        level < depth && println(io, "\\end{itemize}")
+        level > depth && (println(io, "\\begin{itemize}"); needs_end = true)
```

Comment:
I don't recall if julia exports it, but IIRC there are some neat withXXX latex methods, which wrap begins/ends in a slightly cleaner way than printing (though it requires more indenting).